Data Structures/Linked-Lists/Implementations of single linked list.cpp

```diff
@@ -0,0 +1,379 @@
+struct node *addBefore(struct node *start,int data,int item );     //FUNCTION TO ADD A NODE BEFORE A PARTCULAR NODE
+struct node *addAtPos(struct node *start,int data,int pos);        //FUNCTION TO ADD A NODE AT A PARTCULAR NODE
+struct node *del(struct node *start,int data);                     //FUNCTION TO DELETE A NODE
+struct node *reverse(struct node *start);                          //FUNCTION TO REVERSE A NODE
+ 
+//MAIN FUNCTIOM
+
+int main()
+{
+	struct node *start=NULL;	
+	int choice,data,item,pos;
+		
+	while(1)
+	{
+		printf("1.Create List\n");
+		printf("2.Display\n");
+		printf("3.Count\n");
+		printf("4.Search\n");
+		printf("5.Add to empty list / Add at beginning\n");
+		printf("6.Add at end\n");
+		printf("7.Add after node\n");
+		printf("8.Add before node\n");
+		printf("9.Add at position\n");
+		printf("10.Delete\n");
+		printf("11.Reverse\n");
+		printf("12.Quit\n\n");
+		
+		printf("Enter your choice : ");
+		scanf("%d",&choice);
+		
+		switch(choice)
+		{
+		 case 1:
+			start=createList(start);
+			break;
+		 case 2:
+			display(start);
+			break;
+		 case 3:
+			count(start);
+			break;
+		 case 4:
+			printf("Enter the element to be searched : ");
+			scanf("%d",&data);
+			search(start,data);
+			break;
+		 case 5:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			start=addAtBeg(start,data);
+			break;
+		 case 6:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			start=addAtEnd(start,data);
+			break;
+		 case 7:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the element after which to insert : ");
+			scanf("%d",&item);
+			start=addAfter(start,data,item);
+			break;
+		 case 8:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the element before which to insert : ");
+			scanf("%d",&item);
+			start=addBefore(start,data,item);
+			break;
+		 case 9:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the position at which to insert : ");
+			scanf("%d",&pos);
+			start=addAtPos(start,data,pos);
+			break;
+		 case 10:
+			printf("Enter the element to be deleted : ");
+			scanf("%d",&data);
+			start=del(start, data);	
+			break;
+		 case 11:
+			start=reverse(start);
+			break;
+		 case 12:
+			 exit(1);
+		 default:
+			 printf("Wrong choice\n");
+		}/*End of switch */
+	}/*End of while */
+
+return 0;
+
+}
+
+/*End of main()*/
+
+
+struct node *createList(struct node *start)
+{
+	int i,n,data;
+	printf("Enter the number of nodes : ");
+	scanf("%d",&n);
+	start=NULL;
+	if(n==0)
+		return start;
+
+	printf("Enter the element to be inserted : ");
+	scanf("%d",&data);
+	start=addAtBeg(start,data);
+
+	for(i=2;i<=n;i++)
+	{
+		printf("Enter the element to be inserted : ");
+		scanf("%d",&data);
+		start=addAtEnd(start,data);	
+	}
+	return start;
+}
+
+/*End of create_list()*/
+
+

```

Comment:
Keep a uniform number of blank lines. 1 after a function ends, 1 for the comment and 1 following it. Remove the extra from everywhere.

Reply:
I remove all blanks.No blanks for function end and no space for comment.